File: Strivers/src/basicrecursion/sumofseries/SumOfSeries.java
package basicrecursion.sumofseries;

import java.util.Random;

public class SumOfSeries {
    public static void main(String[] args) {
        for (int i = 1; i <= 5; i++) {
            Random rand = new Random();
            Long num = rand.nextLong(i);
            System.out.printf("%d = %-15s%d \n",num, "sum of series ", sumOfSeries(num));
        }
    }

    private static Long sumOfSeries(Long num) {
        if (num <= 0) return 0L;
        return num + sumOfSeries(num - 1);
    }
}

File: Strivers/src/basicrecursion/sumofseries/SumOfSeries.kt
package basicrecursion.sumofseries

fun main(args: Array<String>) {
    for (i in 1..5) {
        val num = (1..5).random().toLong()
        print("$num = ")
        println(sumOfSeries(num))
    }
}

fun sumOfSeries(num: Long): Long {
    if (num <= 0) return 0
    return Math.pow(num.toDouble(), 3.0).toLong() + sumOfSeries(num - 1)
}